tests/e2e/paths/search/granules/granules.spec.js

```diff
@@ -0,0 +1,1180 @@
+            'access-control-expose-headers': 'cmr-hits',
+            'cmr-hits': cmrHits
+          }
+        })
+      })
+
+      await page.route('**/search/granules/timeline', (route) => {
+        const request = route.request()
+        const body = request.postData()
+        if (body) {
+          expect(body).toBe('end_date=2023-12-01T00:00:00.000Z&interval=day&start_date=2018-12-01T00:00:00.000Z&concept_id[]=C1214470488-ASF')
+        }
+
+        route.fulfill({
+          body: JSON.stringify(readableGranuleNameTimelineBody),
+          headers: readableGranuleNameTimelineHeaders
+        })
+      })
+
+      await page.route('**/api', (route) => {
+        const requestBody = JSON.parse(route.request().postData())
+        const expectedBody = JSON.parse(graphQlGetCollection(conceptId))
+
+        expect(requestBody).toEqual(expectedBody)
+        route.fulfill({
+          body: JSON.stringify(readableGranuleNameGraphQlBody),
+          headers: readableGranuleNameGraphQlHeaders
+        })
+      })
+
+      await page.goto('/search/granules?p=C1214470488-ASF&pg[0][id]=S1A_S3_SLC__1SDH_20140615T034444_20140615T034512_001055_00107C_16F1')
+
+      // Ensure the correct number of results were loaded
+      await testResultsSize(page, cmrHits)
+
+      // Readable granule name input is populated
+      await expect(page.getByTestId('granule-filters__readable-granule-name')).toHaveValue('S1A_S3_SLC__1SDH_20140615T034444_20140615T034512_001055_00107C_16F1')
+    })
+  })
+
+  test.describe('When the path is loaded with granule temporal', () => {
+    test.describe('When the temporal range is not recurring', () => {
+      test('loads with the temporal fields populated', async ({ page }) => {
+        const conceptId = 'C1214470488-ASF'
+        const cmrHits = 17231
+
+        await page.route('**/search/granules.json', (route) => {
+          const request = route.request()
+          const body = request.postData()
+          expect(body).toBe('echo_collection_id=C1214470488-ASF&page_num=1&page_size=20&temporal=2020-01-01T00:00:00.000Z,2020-01-31T23:59:59.999Z')
+          route.fulfill({
+            body: JSON.stringify(temporalGranulesBody),
+            headers: {
+              ...commonGranulesHeaders,
+              'access-control-expose-headers': 'cmr-hits',
+              'cmr-hits': cmrHits
+            }
+          })
+        })
+
+        await page.route('**/search/granules/timeline', (route) => {
+          const request = route.request()
+          const body = request.postData()
+          if (body) {
+            expect(body).toBe('end_date=2023-12-01T00:00:00.000Z&interval=day&start_date=2018-12-01T00:00:00.000Z&concept_id[]=C1214470488-ASF')
+          }
+
+          route.fulfill({
+            body: JSON.stringify(temporalTimelineBody),
+            headers: temporalTimelineHeaders
+          })
+        })
+
+        await page.route('**/api', (route) => {
+          const request = route.request()
```

Comment:
```suggestion
 const requestBody = JSON.parse(route.request().postData())
          const expectedBody = JSON.parse(graphQlGetCollection(conceptId))
          expect(requestBody).toEqual(expectedBody)```
          
       Can this one be like the others?

Reply:
Can this one be like the others?